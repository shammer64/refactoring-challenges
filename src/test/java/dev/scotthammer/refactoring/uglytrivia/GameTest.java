package dev.scotthammer.refactoring.uglytrivia;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import java.io.ByteArrayOutputStream;
import java.io.PrintStream;

import static org.junit.jupiter.api.Assertions.*;

class GameTest {

    @Test
    void characterizationTest1() {
        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
        System.setOut(new PrintStream(outputStream));
        Game aGame = new Game();
        aGame.add("Chet");
        aGame.add("Sue");

        for (int i = 0; i < 100; i++) {
            aGame.roll(i % 6 + 1);
            if (i % 9 == 7) {
                aGame.wrongAnswer();
            } else {
                aGame.wasCorrectlyAnswered();
            }
        }

        Approvals.verify(outputStream.toString());
    }

    @Test
    void singlePlayerGameNotPlayable() {
        Game aGame = new Game();
        aGame.add("Chet");
        assertFalse(aGame.isPlayable());
    }

    @Test
    void multiPlayerGameNotPlayable() {
        Game aGame = new Game();
        aGame.add("Chet");
        aGame.add("Sue");
        assertTrue(aGame.isPlayable());
    }

    @Test
    void maxSixPlayersAllowedInGame() {
        Game aGame = new Game();
        aGame.add("Chet");
        aGame.add("Sue");
        aGame.add("Fred");
        aGame.add("Wilma");
        aGame.add("Barney");
        aGame.add("Betty");
        assertThrows(IllegalStateException.class, () -> aGame.add("Joy"));
    }


}