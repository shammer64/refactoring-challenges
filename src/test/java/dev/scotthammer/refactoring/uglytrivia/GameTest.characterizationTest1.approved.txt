Chet was added
They are player number 1
Sue was added
They are player number 2
Chet is the current player
They have rolled a 1
Chet's new location is 1
The category is Science
Science Question 0
Answer was correct!!!!
Chet now has 1 Gold Coins.
Sue is the current player
They have rolled a 2
Sue's new location is 2
The category is Sports
Sports Question 0
Answer was correct!!!!
Sue now has 1 Gold Coins.
Chet is the current player
They have rolled a 3
Chet's new location is 4
The category is Pop
Pop Question 0
Answer was correct!!!!
Chet now has 2 Gold Coins.
Sue is the current player
They have rolled a 4
Sue's new location is 6
The category is Sports
Sports Question 1
Answer was correct!!!!
Sue now has 2 Gold Coins.
Chet is the current player
They have rolled a 5
Chet's new location is 9
The category is Science
Science Question 1
Answer was correct!!!!
Chet now has 3 Gold Coins.
Sue is the current player
They have rolled a 6
Sue's new location is 0
The category is Pop
Pop Question 1
Answer was correct!!!!
Sue now has 3 Gold Coins.
Chet is the current player
They have rolled a 1
Chet's new location is 10
The category is Sports
Sports Question 2
Answer was correct!!!!
Chet now has 4 Gold Coins.
Sue is the current player
They have rolled a 2
Sue's new location is 2
The category is Sports
Sports Question 3
Question was incorrectly answered
Sue was sent to the penalty box
Chet is the current player
They have rolled a 3
Chet's new location is 1
The category is Science
Science Question 2
Answer was correct!!!!
Chet now has 5 Gold Coins.
Sue is the current player
They have rolled a 4
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 5
Chet's new location is 6
The category is Sports
Sports Question 4
Answer was correct!!!!
Chet now has 6 Gold Coins.
Sue is the current player
They have rolled a 6
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 1
Chet's new location is 7
The category is Rock
Rock Question 0
Answer was correct!!!!
Chet now has 7 Gold Coins.
Sue is the current player
They have rolled a 2
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 3
Chet's new location is 10
The category is Sports
Sports Question 5
Answer was correct!!!!
Chet now has 8 Gold Coins.
Sue is the current player
They have rolled a 4
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 5
Chet's new location is 3
The category is Rock
Rock Question 1
Question was incorrectly answered
Chet was sent to the penalty box
Sue is the current player
They have rolled a 6
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 1
Chet is getting out of the penalty box
Chet's new location is 4
The category is Pop
Pop Question 2
Answer was correct!!!!
Chet now has 9 Gold Coins.
Sue is the current player
They have rolled a 2
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 3
Chet is getting out of the penalty box
Chet's new location is 7
The category is Rock
Rock Question 2
Answer was correct!!!!
Chet now has 10 Gold Coins.
Sue is the current player
They have rolled a 4
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 5
Chet is getting out of the penalty box
Chet's new location is 0
The category is Pop
Pop Question 3
Answer was correct!!!!
Chet now has 11 Gold Coins.
Sue is the current player
They have rolled a 6
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 1
Chet is getting out of the penalty box
Chet's new location is 1
The category is Science
Science Question 3
Answer was correct!!!!
Chet now has 12 Gold Coins.
Sue is the current player
They have rolled a 2
Sue is not getting out of the penalty box
Question was incorrectly answered
Sue was sent to the penalty box
Chet is the current player
They have rolled a 3
Chet is getting out of the penalty box
Chet's new location is 4
The category is Pop
Pop Question 4
Answer was correct!!!!
Chet now has 13 Gold Coins.
Sue is the current player
They have rolled a 4
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 5
Chet is getting out of the penalty box
Chet's new location is 9
The category is Science
Science Question 4
Answer was correct!!!!
Chet now has 14 Gold Coins.
Sue is the current player
They have rolled a 6
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 1
Chet is getting out of the penalty box
Chet's new location is 10
The category is Sports
Sports Question 6
Answer was correct!!!!
Chet now has 15 Gold Coins.
Sue is the current player
They have rolled a 2
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 3
Chet is getting out of the penalty box
Chet's new location is 1
The category is Science
Science Question 5
Answer was correct!!!!
Chet now has 16 Gold Coins.
Sue is the current player
They have rolled a 4
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 5
Chet is getting out of the penalty box
Chet's new location is 6
The category is Sports
Sports Question 7
Question was incorrectly answered
Chet was sent to the penalty box
Sue is the current player
They have rolled a 6
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 1
Chet is getting out of the penalty box
Chet's new location is 7
The category is Rock
Rock Question 3
Answer was correct!!!!
Chet now has 17 Gold Coins.
Sue is the current player
They have rolled a 2
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 3
Chet is getting out of the penalty box
Chet's new location is 10
The category is Sports
Sports Question 8
Answer was correct!!!!
Chet now has 18 Gold Coins.
Sue is the current player
They have rolled a 4
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 5
Chet is getting out of the penalty box
Chet's new location is 3
The category is Rock
Rock Question 4
Answer was correct!!!!
Chet now has 19 Gold Coins.
Sue is the current player
They have rolled a 6
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 1
Chet is getting out of the penalty box
Chet's new location is 4
The category is Pop
Pop Question 5
Answer was correct!!!!
Chet now has 20 Gold Coins.
Sue is the current player
They have rolled a 2
Sue is not getting out of the penalty box
Question was incorrectly answered
Sue was sent to the penalty box
Chet is the current player
They have rolled a 3
Chet is getting out of the penalty box
Chet's new location is 7
The category is Rock
Rock Question 5
Answer was correct!!!!
Chet now has 21 Gold Coins.
Sue is the current player
They have rolled a 4
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 5
Chet is getting out of the penalty box
Chet's new location is 0
The category is Pop
Pop Question 6
Answer was correct!!!!
Chet now has 22 Gold Coins.
Sue is the current player
They have rolled a 6
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 1
Chet is getting out of the penalty box
Chet's new location is 1
The category is Science
Science Question 6
Answer was correct!!!!
Chet now has 23 Gold Coins.
Sue is the current player
They have rolled a 2
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 3
Chet is getting out of the penalty box
Chet's new location is 4
The category is Pop
Pop Question 7
Answer was correct!!!!
Chet now has 24 Gold Coins.
Sue is the current player
They have rolled a 4
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 5
Chet is getting out of the penalty box
Chet's new location is 9
The category is Science
Science Question 7
Question was incorrectly answered
Chet was sent to the penalty box
Sue is the current player
They have rolled a 6
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 1
Chet is getting out of the penalty box
Chet's new location is 10
The category is Sports
Sports Question 9
Answer was correct!!!!
Chet now has 25 Gold Coins.
Sue is the current player
They have rolled a 2
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 3
Chet is getting out of the penalty box
Chet's new location is 1
The category is Science
Science Question 8
Answer was correct!!!!
Chet now has 26 Gold Coins.
Sue is the current player
They have rolled a 4
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 5
Chet is getting out of the penalty box
Chet's new location is 6
The category is Sports
Sports Question 10
Answer was correct!!!!
Chet now has 27 Gold Coins.
Sue is the current player
They have rolled a 6
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 1
Chet is getting out of the penalty box
Chet's new location is 7
The category is Rock
Rock Question 6
Answer was correct!!!!
Chet now has 28 Gold Coins.
Sue is the current player
They have rolled a 2
Sue is not getting out of the penalty box
Question was incorrectly answered
Sue was sent to the penalty box
Chet is the current player
They have rolled a 3
Chet is getting out of the penalty box
Chet's new location is 10
The category is Sports
Sports Question 11
Answer was correct!!!!
Chet now has 29 Gold Coins.
Sue is the current player
They have rolled a 4
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 5
Chet is getting out of the penalty box
Chet's new location is 3
The category is Rock
Rock Question 7
Answer was correct!!!!
Chet now has 30 Gold Coins.
Sue is the current player
They have rolled a 6
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 1
Chet is getting out of the penalty box
Chet's new location is 4
The category is Pop
Pop Question 8
Answer was correct!!!!
Chet now has 31 Gold Coins.
Sue is the current player
They have rolled a 2
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 3
Chet is getting out of the penalty box
Chet's new location is 7
The category is Rock
Rock Question 8
Answer was correct!!!!
Chet now has 32 Gold Coins.
Sue is the current player
They have rolled a 4
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 5
Chet is getting out of the penalty box
Chet's new location is 0
The category is Pop
Pop Question 9
Question was incorrectly answered
Chet was sent to the penalty box
Sue is the current player
They have rolled a 6
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 1
Chet is getting out of the penalty box
Chet's new location is 1
The category is Science
Science Question 9
Answer was correct!!!!
Chet now has 33 Gold Coins.
Sue is the current player
They have rolled a 2
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 3
Chet is getting out of the penalty box
Chet's new location is 4
The category is Pop
Pop Question 10
Answer was correct!!!!
Chet now has 34 Gold Coins.
Sue is the current player
They have rolled a 4
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 5
Chet is getting out of the penalty box
Chet's new location is 9
The category is Science
Science Question 10
Answer was correct!!!!
Chet now has 35 Gold Coins.
Sue is the current player
They have rolled a 6
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 1
Chet is getting out of the penalty box
Chet's new location is 10
The category is Sports
Sports Question 12
Answer was correct!!!!
Chet now has 36 Gold Coins.
Sue is the current player
They have rolled a 2
Sue is not getting out of the penalty box
Question was incorrectly answered
Sue was sent to the penalty box
Chet is the current player
They have rolled a 3
Chet is getting out of the penalty box
Chet's new location is 1
The category is Science
Science Question 11
Answer was correct!!!!
Chet now has 37 Gold Coins.
Sue is the current player
They have rolled a 4
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 5
Chet is getting out of the penalty box
Chet's new location is 6
The category is Sports
Sports Question 13
Answer was correct!!!!
Chet now has 38 Gold Coins.
Sue is the current player
They have rolled a 6
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 1
Chet is getting out of the penalty box
Chet's new location is 7
The category is Rock
Rock Question 9
Answer was correct!!!!
Chet now has 39 Gold Coins.
Sue is the current player
They have rolled a 2
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 3
Chet is getting out of the penalty box
Chet's new location is 10
The category is Sports
Sports Question 14
Answer was correct!!!!
Chet now has 40 Gold Coins.
Sue is the current player
They have rolled a 4
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 5
Chet is getting out of the penalty box
Chet's new location is 3
The category is Rock
Rock Question 10
Question was incorrectly answered
Chet was sent to the penalty box
Sue is the current player
They have rolled a 6
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 1
Chet is getting out of the penalty box
Chet's new location is 4
The category is Pop
Pop Question 11
Answer was correct!!!!
Chet now has 41 Gold Coins.
Sue is the current player
They have rolled a 2
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 3
Chet is getting out of the penalty box
Chet's new location is 7
The category is Rock
Rock Question 11
Answer was correct!!!!
Chet now has 42 Gold Coins.
Sue is the current player
They have rolled a 4
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 5
Chet is getting out of the penalty box
Chet's new location is 0
The category is Pop
Pop Question 12
Answer was correct!!!!
Chet now has 43 Gold Coins.
Sue is the current player
They have rolled a 6
Sue is not getting out of the penalty box
Chet is the current player
They have rolled a 1
Chet is getting out of the penalty box
Chet's new location is 1
The category is Science
Science Question 12
Answer was correct!!!!
Chet now has 44 Gold Coins.
Sue is the current player
They have rolled a 2
Sue is not getting out of the penalty box
Question was incorrectly answered
Sue was sent to the penalty box
Chet is the current player
They have rolled a 3
Chet is getting out of the penalty box
Chet's new location is 4
The category is Pop
Pop Question 13
Answer was correct!!!!
Chet now has 45 Gold Coins.
Sue is the current player
They have rolled a 4
Sue is not getting out of the penalty box
